scripts/issue/354/upgrade_remaining_setts_pause.py

```diff
@@ -0,0 +1,315 @@
+
+        safe.post_safe_tx()
+
+    else:
+        for key in SETTV1_KEYS + SETTV1_1_KEYS + SETTV4_KEYS:
+            execute_timelock(
+                safe.badger.timelock,
+                "data/badger/timelock/upgrade_remaining_setts/",
+                key,
+                simulation,
+            )
+
+        # Executing all upgrades together
+        safe.post_safe_tx()
+
+
+def execute_timelock(timelock, queueTx_dir, key, simulation):
+    if simulation == "false":
+        console.print(f"Processing upgrade and patch for {key}...")
+        path = os.path.dirname(queueTx_dir)
+        directory = os.fsencode(path)
+
+        sett = interface.ISettV4h(vaults[key])
+
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if not filename.endswith(".json"):
+                continue
+            txHash = filename.replace(".json", "")
+
+            with open(f"{queueTx_dir}{filename}") as f:
+                tx = json.load(f)
+
+            # Get Sett address from upgrade Tx
+            target_sett = web3.toChecksumAddress("0x" + tx["data"][24:64])
+
+            if sett.address == target_sett:
+                if timelock.queuedTransactions(txHash) == True:
+
+                    # Grab all variables before upgrade
+                    prev_available = sett.available()
+                    prev_gov = sett.governance()
+                    prev_keeper = sett.keeper()
+                    prev_token = sett.token()
+                    prev_controller = sett.controller()
+                    prev_balance = sett.balance()
+                    prev_min = sett.min()
+                    prev_max = sett.max()
+                    prev_getPricePerFullShare = sett.getPricePerFullShare()
+                    prev_strategist = sett.strategist()
+
+                    # Executing upgrade
+                    console.print(f"[green]Executing tx with parameters:[/green] {tx}")
+                    timelock.executeTransaction(
+                        tx["target"], 0, tx["signature"], tx["data"], tx["eta"]
+                    )
+
+                    # Checking all variables are as expected
+                    assert prev_available == sett.available()
+                    assert prev_gov == sett.governance()
+                    assert prev_keeper == sett.keeper()
+                    assert prev_token == sett.token()
+                    assert prev_controller == sett.controller()
+                    assert prev_balance == sett.balance()
+                    assert prev_min == sett.min()
+                    assert prev_max == sett.max()
+                    assert prev_getPricePerFullShare == sett.getPricePerFullShare()
+
+                    if tx["signature"] == UPGRADE_SIG:
+                        assert prev_strategist == sett.strategist()
+                    elif tx["signature"] == SET_STRATEGIST_SIG:
+                        assert TECH_OPS == sett.strategist()
+
```

Comment:
Somewhere around here you could add the set strategist function call so that they get called atomically for each Sett that was upgraded:
```
sett.setStrategist(TECH_OPS, {"from": safe})
```